app/models/schema_version.rb

```diff
@@ -23,7 +23,27 @@ class SchemaVersion < ApplicationRecord
   scope :for_schema,
         ->(schema_id) { where(schema_id: schema_id) }
   scope :for_schema_fingerprint,
-        ->(fingerprint) { joins(:schema).where('schemas.fingerprint = ?', fingerprint) }
+        ->(fingerprint) do
```

Comment:
Should this be pushed into a scope on the `Schema` class since it's really dealing with the internals of that model? Same question on the `for_schema_json` scope. This scope could also be leveraged by `Schema.existing_schema` too.